### PR TITLE
Added Manager For DoorCode

### PR DESCRIPTION
**Pull request by SlicerBX**

### CODE REVIEW

**Comment by SlicerBX** [comment]
Integrated previous implemented backend with newly added front end for managing door codes from admin.

Testing:

1. Login into your admin account. Once logged in, go to the admin dashboard located under the account name on the right of navbar. Once there, click on Door Codes on the nav bar, you should see the following:
Screen Shot 2020-12-19 at 5 49 33 PM

2. To add a door code, click on the + button and fill out the text fields (Door Code and Expire Date). Note that a door code can only be a sequence of seven numbers with a dash after the third number (i.e. ###-####). Note that you can add more than one door code by creating a new line in the text box. Click on the Add Door Codes button to add them to the table manager.
Screen Shot 2020-12-19 at 6 00 49 PM

Screen Shot 2020-12-19 at 6 01 56 PM

3. To edit a single door code, click on the edit icon associated with the door code and input your new door code sequence and expire date. Click on the edit button once all fields are inputted.
Screen Shot 2020-12-19 at 6 02 26 PM

4. To delete a door code, click on the trash icon associated with the door code. A warning message will pop up. To verify the deletion, click yes and the table will update with the deleted door code removed.
Screen Shot 2020-12-19 at 6 02 44 PM

5. To delete all door codes that are expired, click on the clock and trash icon. A warning message will pop up. To verify, click Delete All and the table will update by removing any door codes that have passed their expiration date.
Screen Shot 2020-12-19 at 6 02 54 PM

**Review by IsitaB — COMMENTED** [review]
form validation: it doesn't add doorcodes at all if the formatting isn't right. maybe have an error message

consider letting the user input a number without dashes; render the doorcode with dashes tho

File: api/main_endpoints/routes/DoorCode.js [hunk 1]
```diff
@@ -30,13 +30,62 @@ router.get('/getDoorCodes', (req, res) => {
...
+      res.status(OK).send({doorCode});
+    } else {
+      res.status(NOT_FOUND).send({ message: 'No codes found.' });
+    }
+  }).catch(() => {
+    res.status(NOT_FOUND).send({ message: 'No codes found.' });
+  });
+});
+
+// router.post('/addCode', (req, res) => {
```
Comment (aashrav) [inline api/main_endpoints/routes/DoorCode.js:67]: remove these comments

File: src/Pages/DoorCode/DoorCode.js [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+  async callDoorCodes() {
+    const apiResponse = await getAllDoorCodes(this.state.authToken);
+
+    if (!apiResponse.error) this.setState({
+      doorCodes: apiResponse.responseData });
+  }
+
+  updateQuery(value) {
+    // taking care of empty values
+    // value = typeof value === 'undefined' ? '' : value;
```
Comment (aashrav) [inline src/Pages/DoorCode/DoorCode.js:69]: remove comment

File: src/Pages/DoorCode/SVG.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        8.318-8.31-8.203-3.666 3.666 8.321
+        8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z'
+      />
+    </svg>
+  );
+}
+
+export function sunIcon() {
+  return (
+    // paste the <svg> and body from the website
```
Comment (aashrav) [inline src/Pages/DoorCode/SVG.js:125]: remove comment

File: src/Pages/Overview/OverviewProfile.js [hunk 4]
```diff
@@ -29,13 +31,24 @@ export default function OverviewProfile(props) {
   function mark(bool) {
     return bool ? svg.checkMark() : svg.xMark();
   }
+  useEffect(() => {
+    // All the getDoor() logic in here
```
Comment (aashrav) [inline src/Pages/Overview/OverviewProfile.js:35]: remove comment

File: src/Pages/Profile/admin/AdminView.jsx [hunk 5]
```diff
@@ -101,12 +97,39 @@ export default function Editor(props) {
     { value: 2, name: '2 semesters' }
   ];
 
+  useEffect(() => {
+    // All the getDoor() logic in here
```
Comment (aashrav) [inline src/Pages/Profile/admin/AdminView.jsx:101]: remove

File: src/Pages/Profile/admin/AdminView.jsx [hunk 6]
```diff
@@ -101,12 +97,39 @@ export default function Editor(props) {
...
+      let data = await getPersonsDoorCode(user.email, props.token);
+      if(!data.error){
+        setDoorCode(data.responseData.doorCode.doorCode);
+      }
+    }
+    fetchDoorCode();
+  }, [user.email, props.token]);
+
+  useEffect(() => {
+    // All the getDoor() logic in here
```
Comment (aashrav) [inline src/Pages/Profile/admin/AdminView.jsx:113]: remove

**Review by aashrav — CHANGES_REQUESTED** [review]
Remove comments an rebase

**Review by aashrav — APPROVED** [review]
awesome!

**Review by IsitaB — CHANGES_REQUESTED** [review]
Please have the MemberView Profile page render the doorcode in the Profile doorcode block

Form validation por favor

**Comment by aashrav** [comment]
> doorcode

In order to make this change make sure to edit MemberView/Profile.js...Currently they are trying to access the doorcode attribute from the User schema(which we took out). Instead you will need to call the getPersonsDoorCode() function and pass the email to get the doorcode. In addition, the form validation doesn't work...when I try to create a new doorcode without a date then it throws an error

**Review by aashrav — CHANGES_REQUESTED** [review]
In order to make this change make sure to edit MemberView/Profile.js...Currently they are trying to access the doorcode attribute from the User schema(which we took out). Instead you will need to call the getPersonsDoorCode() function and pass the email to get the doorcode. In addition, the form validation doesn't work...when I try to create a new doorcode without a date then it throws an error

**Pull request closed without merging**